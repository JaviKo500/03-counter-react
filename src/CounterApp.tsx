interface Props {
  value: number;
}

export const CounterApp = ( { value } : Props ) => {
  return (
    <>
      <h1>Counter App</h1>
      <h2>{ value }</h2>
    </>
  )
}
