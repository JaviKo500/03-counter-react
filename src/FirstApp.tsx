const person = {
  name: 'Javiko500',
  age: 30,
  city: 'Santiago',
};

const currentYear = () => {
  return new Date().getFullYear();
}
let counter = 0;
const add = () => {
  return counter++;
}

export const FirstApp = () => {
  return (
    <>
      <h1>
        Hello
      </h1>
      <p>I'm { person.name }</p>
      <p>My age is { person.age } in { currentYear() }</p>
      <p>I live in { person.city }</p>
      <button onClick={ () => add() }>Click me { counter }</button>
      <br />
      <code>{ JSON.stringify(person) }</code>
    </>
  );
}
