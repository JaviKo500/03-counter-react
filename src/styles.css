html, body {
  background-color: #21232A;
  color: #fff;
  font-family: Cambria, Cochin, Georgia, Times, 'Times New Roman', serif;
  font-size: 1.3rem;
  padding: 70px;
}

button {
  padding: 15px;
  font-size: 1.3rem;
  border-radius: 12px;
  background-color: #fff;
  margin-right: 5px;
}