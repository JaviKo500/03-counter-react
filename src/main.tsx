import { App } from './App';
import { StrictMode } from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
  <StrictMode>
    <App />
    <FirstApp title="Hello World" subtitle={'test subtitle'} showTitle />
  </StrictMode>
);