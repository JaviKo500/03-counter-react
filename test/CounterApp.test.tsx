import { render, screen } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';
import React from 'react';

describe('CounterApp.test', () => {
  const initialValue = 100;
  
  test( 'should match snapshot', () => {
    const { container } = render(<CounterApp value={ initialValue } />);
    expect( container ).toMatchSnapshot();
  });
  
  test( `should initial value ${initialValue}`, () => {
    render(<CounterApp value={ initialValue } />);
    expect( screen.getByRole('heading', { level: 2} ).innerHTML ).toContain(initialValue.toString());
  });


});