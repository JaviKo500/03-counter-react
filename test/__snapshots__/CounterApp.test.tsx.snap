// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp.test should match snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    100
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
